libcpuid/libcpuid.h

```diff
@@ -831,7 +832,7 @@ struct msr_driver_t* cpu_msr_driver_open(void);
  *          The error message can be obtained by calling \ref cpuid_error.
  *          @see cpu_error_t
  */
-struct msr_driver_t* cpu_msr_driver_open_core(int core_num);
+struct msr_driver_t* cpu_msr_driver_open_core(uint8_t core_num);
```

Comment:
That would unnecessarily fix the max supported number of cores to 255, for no real savings (you're checking the parameter against `cpuid_get_total_cpus()` anyway; knowing it's within 0..255 doesn't help you at all). And who knows, in the future we might have CPUs with 512 cores or more.
I also don't think that unsigned brings any good (the `x < 0 || x >= cpuid_get_total_cpus()` code is perfect, as it shows your intent more explicitly, and I think the compiler optimizes away the `x < 0` branch check with the same "unsigned compare" trick anyway).

So you should use either `int` or `unsigned` here, but if you want to go with unsigned, just change the sprintf format specifier from `%i` to `%u`.